src/printer.js

```diff
@@ -739,66 +734,69 @@ const expressionKinds = [
   "encapsed",
   "variadic"
 ];
+
+function printLookupNodes(path, options, print) {
+  const node = path.getValue();
+
+  switch (node.kind) {
+    case "propertylookup":
+      return printPropertyLookup(path, options, print);
+    case "staticlookup":
+      return printStaticLookup(path, options, print);
+    case "offsetlookup":
+      return printOffsetLookup(path, options, print);
+    default:
+      return `Have not implemented lookup kind ${node.kind} yet.`;
+  }
+}
+
 function printExpression(path, options, print) {
   const node = path.getValue();
   const lookupKinds = ["propertylookup", "staticlookup", "offsetlookup"];
-  function printLookup(node) {
-    const parent = path.getParentNode();
-    let firstNonMemberParent;
-    let i = 0;
-    do {
-      firstNonMemberParent = path.getParentNode(i);
-      i++;
-    } while (firstNonMemberParent && isMemberish(firstNonMemberParent));
-
-    const shouldInline =
-      (firstNonMemberParent && firstNonMemberParent.kind === "new") ||
-      (node.what.kind === "variable" &&
-        (node.offset &&
-          (node.offset.kind === "constref" ||
-            node.offset.kind === "variable")) &&
-        (parent && !isMemberish(parent)));
 
+  function printLookup(node) {
     switch (node.kind) {
-      case "propertylookup": {
-        return group(
-          concat([
-            path.call(print, "what"),
-            shouldInline
-              ? printPropertyLookup(path, options, print)
-              : group(
-                  indent(
-                    concat([
-                      softline,
-                      printPropertyLookup(path, options, print)
-                    ])
-                  )
-                )
-          ])
-        );
-      }
+      case "propertylookup":
       case "staticlookup":
+      case "offsetlookup": {
```

Comment:
@czosel in future we can other lookup node and better use switch here to avoid rewriting and throw error message in code. This strategy use all prettier languages.